packages/adapter-components/src/definitions/system/fetch/resource.ts

```diff
@@ -13,11 +13,30 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { SaltoError } from '@salto-io/adapter-api'
+import { types } from '@salto-io/lowerdash'
 import { ContextCombinationDefinition, RecurseIntoDefinition } from './dependencies'
 import { AdjustFunction, GeneratedItem, TransformDefinition } from '../shared'
+import { ConfigChangeSuggestion } from '../../user'
 
 export type ResourceTransformFunc = AdjustFunction<{ fragments: GeneratedItem[] }>
 
+type FailEntireFetch = {
+  // If set to true, the entire fetch will fail if an error is encountered while fetching this resource.
+  // By default - false, meaning the fetch will log the error and continue, with only this resource missing.
+  failEntireFetch: boolean
+}
+
+type CustomSaltoError = {
+  customSaltoError: SaltoError
+}
+
+type ConfigSuggestion = {
+  configSuggestion: ConfigChangeSuggestion
+}
+
+type OnErrorHandler = (e: Error) => types.XOR<types.XOR<FailEntireFetch, CustomSaltoError>, ConfigSuggestion>
```

Comment:
1. styling nit: maybe 
```suggestion
type OnErrorHandler = (e: Error) => types.OneOf<{
  // If set to true, the entire fetch will fail if an error is encountered while fetching this resource.
  // By default - false, meaning the fetch will log the error and continue, with only this resource missing.
  failEntireFetch: boolean
  customSaltoError: SaltoError
  configSuggestion: ConfigChangeSuggestion
}>
```
2. might be worth maintaining the special handling for permission errors that we had in the [old infra](https://github.com/salto-io/salto/blob/main/packages/adapter-components/src/elements_deprecated/ducktype/transformer.ts#L455-L458)? and similarly the basic handling for [config suggestions](https://github.com/salto-io/salto/blob/main/packages/adapter-components/src/elements_deprecated/ducktype/transformer.ts#L441-L447)
3. I'm not sure how we combine a default `failEntireFetch = true` with a type override that has, say, a config suggestion. wouldn't we need to override both args?

Reply:
1+2 🚀 🚀  Thanks!!
As for 3 I'm not sure I understand, the field `failEntireFetch` is set to false by default 🤔

Reply:
Got it! That would be true if we had set these fields directly on the resource. 
However, the definition is not: `default.resource.failEntireFetch = true`,
instead, it's: `default.resource.onError = (e) => {...someLogic; return something;}` (in order to be able to define specific actions for specific errors). 
So, when we merge the definitions, the custom one will entirely override the default one :)

Reply:
Discussed f2f - decided to redefine the type in a way that is similar to `ArgsWithCustomizer`